lib/restAPIs/library.js

```diff
@@ -0,0 +1,47 @@
+const assert = require("assert");
+
+const assertParam = require("../util/assertParam");
+const RestUtil = require("../util/RestUtil");
+const buildCommonQuery = require("../util/buildCommonQueryParams");
+
+module.exports = class RestTopic extends RestUtil {
+  _getBase() {
+    return "library";
+  }
+
+  getLibrary(fields = "", outputFlag = "DEFAULT") {
+    const queryParameter = buildCommonQuery(fields);
+
+    return this._makeRequest(
+      `${this._getBase()}`,
+      queryParameter,
+      "GET",
+      null,
+      {
+        outputFlag: outputFlag,
+      }
```

Comment:
outputFlag is a query param, so will need to be part of the object `queryParameter`:
https://demo.microstrategy.com/MicroStrategyLibrary/api-docs/index.html#/Library/getLibrary

```
queryParameter.outputFlag = outputFlag;
```

Reply:
Two ways come to mind. If you want to match up parameters using unique "keys", the function could be changed to take an object instead of multiple parameters:
```javascript
getLibrary({ outputFlag: 'DEFAULT', fields }) {
... definition here ...
}
```

Then instead of calling it with comma-separated parameters, you'd pass an object with the expected keys:
```javascript
library.getLibrary({ outputFlag: 'somethingElse' });
```

Anything not provided will be undefined, so it's a great way to define functions with many parameters (or even just many optional parameters), since the order doesn't matter. Even though we're now passing an object, desctructuring keeps it easy to read. My concern with this, as clean as it is for this example, is that it's not consistent with the rest which may make it less intuitive...

The alternative option is ordering parameters based on which are likely to be changed more often, and passing a `falsy` parameter (such as `false` or `null`) when a parameter should be skipped. Example in your custom implementation:
```javascript
library.getLibrary(false, 'somethingElse');
```

This is the approach I've been taking so far, out of the two immediate ideas that come to mind. I've found it cleaner to make sure the most commonly used parameters are provided first. 

For example, if the implementation is `getLibrary(outputFlag, fields)`, with outputFlag as the first parameter, the following would all bring the same results:
```javascript
//same as getLibrary(undefined, undefined), although first param would default to the string 'DEFAULT'
const res1 = await library.getLibrary();

// would come through as getLibrary('DEFAULT', undefined);
const res2 = await library.getLibrary('DEFAULT');

// would come through as getLibrary('DEFAULT', false); 
const res3 = await library.getLibrary('DEFAULT', false);
```

This means unless you're looking to leverage the fields parameter, you don't need to provide it. 

I've thought so far that "fields" is less likely to be used than some of the parameters that are unique to these methods (at least early on in dev), but I'm open to other ideas with the goal of this being as intuitive as possible with as little "docs" reading as possible!